evm/src/fuzz/strategies/state.rs

```diff
@@ -1,21 +1,27 @@
 use super::fuzz_param_from_state;
-use crate::{executor::StateChangeset, utils};
+use crate::{executor::StateChangeset, fuzz::invariant::FuzzRunIdentifiedContracts, utils};
 use bytes::Bytes;
 use ethers::{
-    abi::Function,
+    abi::{Abi, Function},
+    prelude::ArtifactId,
     types::{Address, Log, H256, U256},
 };
+use foundry_utils::diff_score;
+use parking_lot::RwLock;
 use proptest::prelude::{BoxedStrategy, Strategy};
 use revm::{
     db::{CacheDB, DatabaseRef},
-    opcode, spec_opcode_gas, SpecId,
+    opcode, spec_opcode_gas, Filth, SpecId,
+};
+use std::{
+    collections::{BTreeMap, HashSet},
+    io::Write,
+    sync::Arc,
 };
-use std::{cell::RefCell, collections::HashSet, io::Write, rc::Rc};
-
 /// A set of arbitrary 32 byte data from the VM used to generate values for the strategy.
 ///
 /// Wrapped in a shareable container.
-pub type EvmFuzzState = Rc<RefCell<HashSet<[u8; 32]>>>;
+pub type EvmFuzzState = Arc<RwLock<HashSet<[u8; 32]>>>;
```

Comment:
Just noting here for merge conflict that it is important that this becomes a `BTreeSet`, otherwise the fuzzer is not deterministic when you give it a seed